int32   minX
int32   minY
int32   maxX
int32   maxY
